source/server/admin/logs_handler.cc

```diff
@@ -137,6 +138,11 @@ absl::Status LogsHandler::changeLogLevel(Http::Utility::QueryParams& params) {
       }
 
       name_levels[name] = *level_to_use;
+      if (Logger::Context::useFineGrainLogger()) {
+        ENVOY_LOG(info, "adding fine-grain log update, glob='{}' level='{}'", name,
+                  spdlog::level::level_string_views[*level_to_use]);
+        glob_levels.emplace_back(name, *level_to_use);
```

Comment:
can we possibly have just one map here? seems like for non-fine grain we use only name levels, and for fine grain we populate both `name_level` and `glob_level` with the former unused.

Reply:
I thought this a bit. We still want to keep the order since the design of fine-grain will consider the order for verbosity update, and the first match matters.

Reply:
you're right, we can't use just one map here. But I don't think we need to populate both if we only use one or the other. e.g. we can have something like:

```
if (Logger::Context::useFineGrainLogger()) {
    glob_levels.emplace_back(name, *level_to_use);
} else {
    name_levels[name] = *level_to_use;
}
```

Which would cut number of inserts into some struct for fine grain use case by half.

You could also latch the `useFineGrainLogger() ` outside of the loop since we don't need call that fn every loop iteration given switching from fine grain -> component or vice versa isn't supported.

Reply:
Good catch, done.